Fix notifications timestamp for non PutObject operations

welansari opened the pull request:
We use the lastModified field to set the timestamp of the notifications, however this field is only updated when the object’s data is modified (meaning only on putObject operations).

The change stream entries contains a timestamp, we will use that as a replacement.

Issue: BB-458

Commit:
use mongo operation time as notification timestamp

We use the lastModified field to set the timestamp of the notifications,
however this field is only updated when the object’s data is modified
(meaning only on putObject operations).

Change stream entries contain a field containing the timestamp of the
operation in MongoDB, we will use that instead.

Issue: BB-458

Commit:
update tests

Issue: BB-458

bert-e commented:
# Hello kerkesni,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/backbeat/doc/user).





*Status report is not available.*

bert-e commented:
# Request integration branches



Waiting for integration branch creation to be requested by the user.

To request integration branches, please comment on this pull request with the following command:

```
/create_integration_branches
```

Alternatively, the `/approve` and `/create_pull_requests` commands will automatically
create the integration branches.

Commit:
remove useless $toDate projection

$dateToString returns a date as a string, which
can directly be used to create a JS date

Issue: BB-458

Kerkesni commented:
/approve

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/BB-458-fix-notifications-timestamp` into
`development/8.6`
* [w/8.7/bugfix/BB-458-fix-notifications-timestamp](https://github.com/scality/backbeat/compare/development/8.7...w/8.7/bugfix/BB-458-fix-notifications-timestamp) will be merged into `development/8.7` 



The following branches will **NOT** be impacted:

* `development/7.10`
* `development/7.4`
* `development/7.70`
* `development/8.5`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```







*The following options are set:* **approve**

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/backbeat/commit/ae569e0088dd65933a8137f989c6bad87880dcab) did not succeed in branch w/8.7/bugfix/BB-458-fix-notifications-timestamp.




*The following options are set:* **approve**

bert-e commented:
# Request integration branches



Waiting for integration branch creation to be requested by the user.

To request integration branches, please comment on this pull request with the following command:

```
/create_integration_branches
```

Alternatively, the `/approve` and `/create_pull_requests` commands will automatically
create the integration branches.

welansari commented:
/approve

bert-e commented:
# History mismatch



Merge commit #f691adb3ebbb82f7861686f3cf92b92fcad75dd4 on the integration branch
`w/8.7/bugfix/BB-458-fix-notifications-timestamp` is merging a branch which is neither the current
branch `bugfix/BB-458-fix-notifications-timestamp` nor the development branch
`development/8.7`.

It is likely due to a rebase of the branch `bugfix/BB-458-fix-notifications-timestamp` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **approve**

welansari commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **approve**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/BB-458-fix-notifications-timestamp` into
`development/8.6`
* [w/8.7/bugfix/BB-458-fix-notifications-timestamp](https://github.com/scality/backbeat/compare/development/8.7...w/8.7/bugfix/BB-458-fix-notifications-timestamp) will be merged into `development/8.7` 



The following branches will **NOT** be impacted:

* `development/7.10`
* `development/7.4`
* `development/7.70`
* `development/8.5`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```







*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/8.6`

* :heavy_check_mark: `development/8.7`



The following branches will **NOT** be impacted:

* `development/7.10`
* `development/7.4`
* `development/7.70`
* `development/8.5`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/8.6`

* :heavy_check_mark: `development/8.7`



The following branches have **NOT** changed:

* `development/7.10`
* `development/7.4`
* `development/7.70`
* `development/8.5`



Please check the status of the associated issue BB-458.

Goodbye kerkesni.
